component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/moesif/MoesifReporter.java

```diff
@@ -0,0 +1,76 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.am.analytics.publisher.reporter.moesif;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.am.analytics.publisher.exception.MetricCreationException;
+import org.wso2.am.analytics.publisher.reporter.AbstractMetricReporter;
+import org.wso2.am.analytics.publisher.reporter.CounterMetric;
+import org.wso2.am.analytics.publisher.reporter.MetricSchema;
+import org.wso2.am.analytics.publisher.reporter.TimerMetric;
+import org.wso2.am.analytics.publisher.reporter.moesif.util.MoesifMicroserviceConstants;
+import org.wso2.am.analytics.publisher.retriever.MoesifKeyRetriever;
+import org.wso2.am.analytics.publisher.util.Constants;
+
+import java.util.Map;
+
+/**
+ * Moesif Metric Reporter Implementation. This implementation is responsible for sending analytics data into Moesif
+ * dashboard in a secure and reliable way.
+ */
+public class MoesifReporter extends AbstractMetricReporter {
+    private static final Logger log = LoggerFactory.getLogger(MoesifReporter.class);
+    private final Map<String, String> properties;
+    private final EventQueue eventQueue;
+
+    public MoesifReporter(Map<String, String> properties) throws MetricCreationException {
+        super(properties);
+        this.properties = properties;
+        MoesifKeyRetriever keyRetriever =
+                MoesifKeyRetriever.getInstance(properties.get(MoesifMicroserviceConstants.GA_USERNAME_CONFIG_KEY),
+                        properties.get(MoesifMicroserviceConstants.GA_PWD_CONFIG_KEY));
+        int queueSize = Constants.DEFAULT_QUEUE_SIZE;
+        int workerThreads = Constants.DEFAULT_WORKER_THREADS;
+        if (properties.get(Constants.QUEUE_SIZE) != null) {
+            queueSize = Integer.parseInt(properties.get(Constants.QUEUE_SIZE));
+        }
+        if (properties.get(Constants.WORKER_THREAD_COUNT) != null) {
+            workerThreads = Integer.parseInt(properties.get(Constants.WORKER_THREAD_COUNT));
+        }
+        eventQueue = new EventQueue(queueSize, workerThreads, keyRetriever);
+    }
+
+    @Override
+    protected void validateConfigProperties(Map<String, String> map) throws MetricCreationException {
+
+    }
+
+    @Override
+    public CounterMetric createCounter(String name, MetricSchema metricSchema) throws MetricCreationException {
+        MoesifCounterMetric logCounterMetric = new MoesifCounterMetric(name, eventQueue, metricSchema);
+
+        return logCounterMetric;
```

Comment:
logCounterMetric is unnecessary

Reply:
Ack. 
Rename the variable to something make more sense.